src/Aquifer.Well.API/Services/NotificationHubService.cs

```diff
@@ -0,0 +1,170 @@
+using Aquifer.Well.API.Endpoints.PushNotifications.Models;
+using Microsoft.Extensions.Options;
+using Microsoft.Azure.NotificationHubs;
+
+namespace Aquifer.Well.API.Services;
+
+public class NotificationHubService : INotificationService
+{
+    private readonly NotificationHubClient _hub;
+    private readonly Dictionary<string, NotificationPlatform> _installationPlatform;
+    private readonly ILogger<NotificationHubService> _logger;
+    public NotificationHubService(IOptions<NotificationHubOptions> options, ILogger<NotificationHubService> logger)
+
+    {
+        _logger = logger;
+        _hub = NotificationHubClient.CreateClientFromConnectionString(options.Value.ConnectionString, options.Value.Name);
+
+        _installationPlatform = new Dictionary<string, NotificationPlatform>
+        {
+            { nameof(NotificationPlatform.Apns).ToLower(), NotificationPlatform.Apns },
+            { nameof(NotificationPlatform.FcmV1).ToLower(), NotificationPlatform.FcmV1 }
+        };
+    }
+
+    public async Task<bool> CreateOrUpdateInstallationAsync(DeviceInstallation deviceInstallation, CancellationToken token)
+    {
+        if (string.IsNullOrWhiteSpace(deviceInstallation?.InstallationId) ||
+            string.IsNullOrWhiteSpace(deviceInstallation?.Platform) ||
+            string.IsNullOrWhiteSpace(deviceInstallation?.PushChannel))
+        {
+            return false;
```

Comment:
I'd expect that (a) invalid device installation requests would be caught by an endpoint validator, and then (b) this `public` method can then throw an `ArgumentException` or `ArgumentNullException` if it gets data that it can't handle (which should never happen).